Mathlib/Data/Finset/Sort.lean

```diff
@@ -275,6 +275,19 @@ theorem orderEmbOfCardLe_mem (s : Finset α) {k : ℕ} (h : k ≤ s.card) (a) :
   simp only [orderEmbOfCardLe, RelEmbedding.coe_trans, Finset.orderEmbOfFin_mem,
     Function.comp_apply]
 
+lemma orderEmbOfFin_compl_singleton {n : ℕ} {i : Fin (n + 1)} {k : ℕ}
+    (h : ({i}ᶜ : Finset _).card = k) : ({i}ᶜ : Finset _).orderEmbOfFin h =
+      Fin.succAbove i ∘ Fin.cast (h.symm.trans (by simp [Finset.card_compl])) := by
+  rw [eq_comm]
+  convert Finset.orderEmbOfFin_unique _ (fun x ↦ ?_)
+    ((Fin.strictMono_succAbove _).comp (Fin.cast_strictMono _))
+  simp
```

Comment:
You can state this without a function coercion as
```suggestion
lemma orderEmbOfFin_compl_singleton {n : ℕ} {i : Fin (n + 1)} {k : ℕ}
    (h : ({i}ᶜ : Finset _).card = k) :
    ({i}ᶜ : Finset _).orderEmbOfFin h =
      (Fin.castOrderIso <| by simp_all [Finset.card_compl]).toOrderEmbedding.trans (Fin.succAboveOrderEmb i) := by
  apply DFunLike.coe_injective
  rw [eq_comm]
  convert Finset.orderEmbOfFin_unique _ (fun x ↦ ?_)
    ((Fin.strictMono_succAbove _).comp (Fin.cast_strictMono _))
  simp
```
Perhaps this one is a good simp lemma:
```lean-4
@[simp]
lemma orderEmbOfFin_compl_singleton' {n : ℕ} {i : Fin (n + 1) (h : ({i}ᶜ : Finset _).card = n) :
    ({i}ᶜ : Finset _).orderEmbOfFin h = Fin.succAboveOrderEmb i :=
  orderEmbOfFin_compl_singleton _
```

Reply:
Added a version of that after removing `h` from it (removing `h` should make it more convenient for a backwards rewrite, at least).